Implemented camera movement using the mouse

gd-harco opened the pull request:
- Updated gitignore
- Started work on mouse handling
- Update mouse handling in 'on_loop' function
- Refactor drawing methods for render improvements
- Replace rewrite_img with fill_color in on_loop function
- Refactor header file definitions for bonus code
- removed binary from git cache
- Refactored key press handler and added mouse motion handling
- BROKEN Add mouse movement handling for player rotation
- Reverberate mandatory part change on the color filling on the bonus part.
- bugfixes
- Removed mouse.c
- Refactor mouse handling, add struct for mouse information
- Optimize mouse movement function in hooks_bonus.c
- Added a new file for mouse hooks and refactored code to fit the norm + misc. norme fixes.

gd-harco committed:
Updated gitignore

gd-harco committed:
Started work on mouse handling

A new mouse handling functionality has been included to improve user interaction capabilities. On a mouse-click event, the mouse goes into listening mode - its position is now centered on screen and the cursor gets hidden. This feature can be turned off by pressing 'r' key, which brings the cursor back to normal mode and positions it at the center of the screen.

gd-harco committed:
Update mouse handling in 'on_loop' function

Added the mouse position tracking feature in 'on_loop' function. When the mouse is clicked, its listen state is set to true, the cursor hides and its position is now centered on screen. This allows for improved user interaction by tracking the mouse position and updating player's angle based on mouse movements. Removed a redundant print statement "Mouse listen" for cleaner code.

gd-harco committed:
Refactor drawing methods for render improvements

The previous drawing methods ("draw_wall_slice" and "draw_color_column") were modified to simplify and optimize rendering. The draw color column code is now integrated into a new function "fill_color". This removes multiple draw color column calls in "draw_wall_slice" and centralizes color filling. These changes make the code more concise and efficient by reducing repetitive actions, leading to a potential increase in render speed.

gd-harco committed:
Replace rewrite_img with fill_color in on_loop function

With this commit, we have replaced the `rewrite_img` function with `fill_color` function in the `on_loop` function. The `rewrite_img` function was previously being used to update the image. But in order to make the code more efficient and easier to manage, we have switched to using `fill_color`.

The `fill_color` function also takes in additional arguments for floor and ceiling color. These color values are now obtained from the map data, making it more flexible and reusable. This change should help in improving the performance and maintainability of the image updating process.

Lastly, there was a method "fill_color" declared in the file srcs/render/render.c which is now removed since it's definition is moved to a global header file. This makes the function reusable for the rest of the project.

gd-harco committed:
Refactor header file definitions for bonus code

The definitions for bonus related header files have been updated in this commit. Originally, the headers used general names such as 'player.h'. These have been renamed with prefixes to reflect that they are associated with the bonus part of the application, for example 'player_bonus.h'.

This change improves the clarity of the codebase and ensures easier navigation and scalability for future updates. It would be beneficial to separate the core and the bonus functionality, which could have different behaviors, hence the explicit naming for bonus headers. It also enhances the maintainability and code readability by making the file responsibilities more explicit.

gd-harco committed:
removed binary from git cache

gd-harco committed:
Refactored key press handler and added mouse motion handling

The on_key_press function block indentation has been standardized for readability. A mouse.c file was added to handle mouse motion when mouse_listen is true. This addition improves the user interface by introducing a new way to control the player's movement.

gd-harco committed:
BROKEN Add mouse movement handling for player rotation

The mouse movement handling was introduced to improve the user interface for controlling the player's rotation. The feature has been implemented in a new 'mouse_mouvement' function and linked with 'mlx_hook' in main_bonus.c. The on_loop function has been cleaned up to remove unnecessary code related to the old way of handling mouse input. Unnecessary print statement in mouse.c has also been removed.

gd-harco committed:
Reverberate mandatory part change on the color filling on the bonus part.

gd-harco committed:
Merge branch 'ceiling_floor_opti' into mouse_mouvement

gd-harco committed:
bugfixes

gd-harco committed:
Removed mouse.c

gd-harco committed:
Refactor mouse handling, add struct for mouse information

Renamed 'mouse_listen' to 'mouse.listen' and introduced 'mouse' struct to encapsify mouse properties including x, y coordinates and mouse listening status. This change is made to improve code clarity and encapsulation. Also, this eliminates need of 'mouse_pos' object in 'mouse_mouvement', improving efficiency. Removed the now redundant 'mouse.c'.

gd-harco committed:
Optimize mouse movement function in hooks_bonus.c

Adjusted the function mouse_movement which is used to update player rotation. Previously, the player rotation was not being updated despite providing suitable values. Debugging-related prints statement and an unnecessary comment have been removed for code cleanup. Furthermore, player's angle change was adjusted by a factor of 0.05 to get a smoother rotation experience.

gd-harco committed:
Added a new file for mouse hooks and refactored code to fit the norm + misc. norme fixes.

A new file, mouse_hooks_bonus.c, was added to handle mouse events. Previously, mouse-related functions were mixed with keyboard event handlers in hooks_bonus.c causing confusion. This change makes the code more maintainable with clear responsibility segregation. Refactored some function names related to mouse events for better clarity. The changes were also reflected in the Makefile and respective header files.

gd-harco committed:
Change mouse button listening from button 2 to 3

In `mouse_hooks_bonus.c`, the listen trigger for mouse actions was changed from button 2 to button 3 to better align with common user interfaces. This change is expected to enhance user experience by conforming to standard navigation behavior in most operating systems where the right mouse button (usually button 3) is used for context-based actions.

BenoitRoux0 commented on bonus/srcs/mouse_hooks_bonus.c:
```diff
@@ -0,0 +1,43 @@
+	if (data->mouse.listen)
+	{
+		mlx_mouse_get_pos(data->mlx, data->win, &data->mouse.x, &data->mouse.y);
+		data->player.angle += (data->mouse.x - WIN_WIDTH / 2) * 0.05;
```
Tu peux directement utiliser la coordonnees x donnee en parametre pour fluidifier les mouvements et eviter les ralentissements causes par les appels a mlx_mouse_get_pos.

gd-harco committed:
No longer using mlx_mouse_get_pos()

The mouse_hooks_bonus.c file has been changed to improve gameplay interaction. The player's angle calculation was tweaked and direct assignments of mouse coordinates have been made instead of getting them from MLX library. These changes are meant to streamline gameplay and reduce dependence on the MLX library's mouse functions.

gd-harco committed:
Add window focus check for mouse movements

Modified the mouse movement handler to only record movements if the associated window is currently in focus. This was achieved by adding a window focus condition in on_movement function in mouse_hooks_bonus.c, and two new functions, on_focus_in and on_focus_out to handle status changes of window focus. These changes will reduce unwanted mouse interferences from other windows and improve the overall interactivity of the application.

BenoitRoux0 reviewed (approved):
Le mouvement est fluide et la souris est bien ignorée quand le focus de l'utilisateur quitte la fenetre.
